crypto/ex_data.c

```diff
@@ -24,22 +25,21 @@ int do_ex_data_init(OPENSSL_CTX *ctx)
 /*
  * Return the EX_CALLBACKS from the |ex_data| array that corresponds to
  * a given class.  On success, *holds the lock.*
+ * The |global| parameter is assumed to be non null (checked by the caller).
  */
-static EX_CALLBACKS *get_and_lock(OPENSSL_CTX *ctx, int class_index)
+static EX_CALLBACKS *get_and_lock(OSSL_EX_DATA_GLOBAL *global, int class_index)
 {
     EX_CALLBACKS *ip;
-    OSSL_EX_DATA_GLOBAL *global = NULL;
 
     if (class_index < 0 || class_index >= CRYPTO_EX_INDEX__COUNT) {
         CRYPTOerr(CRYPTO_F_GET_AND_LOCK, ERR_R_PASSED_INVALID_ARGUMENT);
         return NULL;
     }
 
-    global = openssl_ctx_get_ex_data_global(ctx);
-    if (global == NULL || global->ex_data_lock == NULL) {
+    if (global->ex_data_lock == NULL) {
         /*
-         * If we get here, someone (who?) cleaned up the lock, so just
-         * treat it as an error.
+         * If we get here, someone cleaned up the lock, so just treat
```

Comment:
why did you remove the fact that we don't know who?  It's arguably a bug/TBD/TODO to find out.

Reply:
I believed it was obsolete. Do you suggest restoring it?

Reply:
unless someone else on the project agrees with you, I think it should be restored.

Reply:
Restored